pkg/server/chaosd/redis.go

```diff
@@ -121,3 +125,69 @@ func (s *Server) recoverSentinelStop(attack *core.RedisCommand) error {
 	}
 	return nil
 }
+
+func (s *Server) expireAllKeys(attack *core.RedisCommand, cli *redis.Client) error {
+	// Get all keys from the server
+	allKeys, err := cli.Keys(cli.Context(), "*").Result()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	expiration, err := time.ParseDuration(attack.Expiration)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if attack.Option == "NX" {
+		for _, key := range allKeys {
+			result, err := cli.ExpireNX(cli.Context(), key, expiration).Result()
```

Comment:
How about defining a variable `ExpireFunc`, and setting it to a specified function depending on the `attack.Option`.  For example, if the option is "NX", set it to `cli.ExpireNX`. In this way, we can remove some redundant codes.